### PR TITLE
Rewrite system tests for ML Engine service

### PR DESCRIPTION
**Pull request by MrGeorgeOwl**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/system/providers/google/cloud/ml_engine/example_mlengine.py [hunk 1]
```diff
@@ -37,70 +40,74 @@
...
-MODEL_NAME = os.environ.get("GCP_MLENGINE_MODEL_NAME", "model_name")
-
-SAVED_MODEL_PATH = os.environ.get("GCP_MLENGINE_SAVED_MODEL_PATH", "gs://INVALID BUCKET NAME/saved-model/")
-JOB_DIR = os.environ.get("GCP_MLENGINE_JOB_DIR", "gs://INVALID BUCKET NAME/keras-job-dir")
-PREDICTION_INPUT = os.environ.get(
-    "GCP_MLENGINE_PREDICTION_INPUT", "gs://INVALID BUCKET NAME/prediction_input.json"
-)
+PROJECT_ID = os.environ.get("GCP_PROJECT_ID")
+ENV_ID = os.environ.get("SYSTEM_TESTS_ENV_ID")
+MODEL_NAME = os.environ.get("GCP_MLENGINE_MODEL_NAME", f"example_mlengine_model_{ENV_ID}")
```
Comment (bhirsz) [inline tests/system/providers/google/cloud/ml_engine/example_mlengine.py]: Just define constant: MODEL_NAME = f"example_mlengine_model_{ENV_ID}"

File: tests/system/providers/google/cloud/ml_engine/example_mlengine.py [hunk 2]
```diff
@@ -37,70 +40,74 @@
...
+BUCKET_NAME = os.environ.get("BUCKET_NAME", f"example_mlengine_bucket_{ENV_ID}")
```
Comment (bhirsz) [inline tests/system/providers/google/cloud/ml_engine/example_mlengine.py]: BUCKET_NAME = f"example_mlengine_bucket_{ENV_ID}"

same for other variables - don't use env variables if it's not necessary.

File: tests/system/providers/google/cloud/ml_engine/example_mlengine.py [hunk 3]
```diff
@@ -37,70 +40,74 @@
...
+        predict_data = generate_model_predict_input_data()
+        with open(path_to_file, 'w') as file:
+            for p in predict_data:
+                file.write(f'{{"input_layer": [{p}]}}\n')
+
+    write_data = PythonOperator(
+        task_id="write-predict-data-file",
+        python_callable=write_predict_file,
+        op_args=(PATH_TO_PREDICT_FILE,),
     )
```
Comment (bhirsz) [inline tests/system/providers/google/cloud/ml_engine/example_mlengine.py]: The code is fine but with the introduction of ``@task`` it's recommended more: https://airflow.apache.org/docs/apache-airflow/stable/howto/operator/python.html

File: tests/system/providers/google/cloud/ml_engine/example_mlengine.py [hunk 4]
```diff
@@ -216,28 +222,38 @@
...
 
     # [START howto_operator_gcp_mlengine_delete_version]
-    delete_version = MLEngineDeleteVersionOperator(
-        task_id="delete-version", project_id=PROJECT_ID, model_name=MODEL_NAME, version_name="v1"
+    delete_version_v1 = MLEngineDeleteVersionOperator(
+        task_id="delete-version-v1",
+        project_id=PROJECT_ID,
+        model_name=MODEL_NAME,
+        version_name="v1",
+        trigger_rule=TriggerRule.ALL_DONE,
```
Comment (bhirsz) [inline tests/system/providers/google/cloud/ml_engine/example_mlengine.py:220]: keep trigger_rule out of doc example - as unnecessary addition:
```
# [START..
operator code..
# [END..
operator.trigger_rule = TriggerRule.ALL_DONE
```

**Review by potiuk — APPROVED** [review]

**Pull request merged**